packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts

```diff
@@ -0,0 +1,15 @@
+import { Flavor } from '@/types/savingsPlanConsumption.type';
+
+export const calculateAverageUsage = (flavors: Flavor[]): number => {
+  const utilizationValues = flavors.flatMap(({ periods }) =>
+    periods
+      .map(({ utilization }) => parseFloat(utilization.replace('%', '')))
+      .filter((value) => !Number.isNaN(value)),
+  );
+
+  const averageUtilization =
+    utilizationValues.reduce((sum, value) => sum + value, 0) /
+    utilizationValues.length;
+
+  return averageUtilization;
+};
```

Comment:
`averageUtilization` constant is useless, you can return directly chained functions :)

```suggestion
export const calculateAverageUsage = (flavors: Flavor[]): number => flavors
    .flatMap(({ periods }) =>
      periods
        .map(({ utilization }) => parseFloat(utilization.replace('%', '')))
        .filter((value) => !Number.isNaN(value)),
    )
    .reduce((sum, value, _, array) => sum + value / array.length, 0);
};
```

Or something like that ;)